src/rpp/rpp/operators/fwd/take.hpp

```diff
@@ -46,13 +46,24 @@ struct member_overload<Type, SpecificObservable, take_tag>
     template<typename...Args>
     auto take(size_t count) const & requires is_header_included<take_tag, Args...>
     {
-        return static_cast<const SpecificObservable*>(this)->template lift<Type>(take_impl<Type>{count});
+        return cast_this()->template lift<Type>(take_impl<Type>{count});
```

Comment:
Then you can return const & instead of const * to use `.lift` instead of `-> template lift` :-)

Reply:
Hmm, this is not up-to-dated code here comparing to up-to-dated code in branch `tcw165:op/take`.